Pi_Code/Controller_Operations.py

```diff
@@ -0,0 +1,113 @@
+from multiprocessing import Process
+import pygame
+
+# Axes legend:
+# Axes 0: Roll (right is positive, left is negative)
+# Axes 1: Pitch (back is positive, forward is negative)
+# Axes 2: Throttle (0% position is positive, 100% position is negative)
+# Axes 3: Yaw (clockwise is positive, counterclockwise is negative)
+# Button legend:
+# Button 0: Trigger
+# Button 6: Labelled as 07 on the joystick
+# Button 5: Labelled as 06 on the joystick
+# Button 1: Labelled as 02 on the joystick
+
+
+class ControllerOps:
+
+    def __init__(self, dataFile, connection, SaveSensorData):
+        # We need to display a window in order to read joystick inputs.
+        pygame.joystick.init()
+        screen = pygame.display.set_mode((10, 10))
+
+        self.controller = pygame.joystick.Joystick(0)
+        self.controller.init()
+
+        self.idle = True
+        self.read_stick = True
+        self.autosub = True
+
+        dataFile.write("Found joystick: " + self.controller.get_name())
+
+        p = Process(target = self.getStickPosition, args = (dataFile, connection, SaveSensorData))
+        p.start()
+
+
+    def getStickPosition(self, connection, dataFile, SaveSensorData):
+        readController = True
+        while (readController):
+
+            while self.idle:
```

Comment:
Please change all the inside whiles to if statemensts, we can't exit the code if it's sutck in the inner loops